File: Data Structures & Algorithms/two-integer-sum/submission-6.swift
// #Plan
// #1. create empty dict
// #2. iterate the list
// #3. get compliment by doing target - num 
// #4. if compliment is in dict return [compliment, curIdx]
// #5. else save compliment
class Solution {
    func twoSum(_ nums: [Int], _ target: Int) -> [Int] {
        // #1. create empty dict
        var numIdxDict : [Int: Int] = [:]
        // #2. iterate the list
        for (idx, num) in nums.enumerated(){
            // #3. get compliment by doing target - num 
            var compliment = target - num

            // #4. if compliment is in dict return [compliment, curIdx]
            if let complimentIdx = numIdxDict[compliment] {
                return [complimentIdx, idx]
            }
            // #5. else save compliment
            numIdxDict[num] = idx
        }
        return []
    }
}

File: Data Structures & Algorithms/find-target-in-rotated-sorted-array/submission-6.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        s = 0
        e = len(nums)-1
        while s <= e:
            mid = (s + e)//2
            if nums[mid] == target:
                return mid
            if nums[s] <= nums[mid]:
                if nums[mid] < target or target < nums[s]:
                    s = mid + 1
                else:
                    e = mid -1
            else:
                if target < nums[mid] or target > nums[e]:
                    e = mid - 1
                else:
                    s = mid + 1
        return -1
